lib/promise.rb

```diff
@@ -54,10 +53,15 @@ def sync
   end
 
   def fulfill(value = nil, backtrace = nil)
-    dispatch(backtrace) do
-      @state = :fulfilled
-      @value = value
+    if Promise === value
```

Comment:
Yup, MRI just uses `return rb_obj_is_kind_of(arg, mod)` and both `kind_of?` and `is_a?` refer to `rb_obj_is_kind_of` internally.